viewer-prototype/src/browser/trace-viewer/trace-viewer.tsx

```diff
@@ -111,52 +111,66 @@ export class TraceViewerWidget extends ReactWidget {
                             break;
                         }
                         const trace = await this.traceManager.openTrace(tracesArray[i].toString(), tracesArray[i].name);
                         if (trace) {
                             traces.push(trace);
+                        } else {
+                            invalidTraces.push(tracesArray[i].name.concat(tracesArray[i].ext));
                         }
                     }
 
                     if (isCancelled.value) {
                         // Rollback traces
-                        progress.report({message: 'Rolling back traces', work: {done: 50, total: 100}});
+                        progress.report({ message: 'Rolling back traces', work: { done: 50, total: 100 } });
                         for (let i = 0; i < traces.length; i++) {
                             await this.traceManager.closeTrace(traces[i].UUID);
                         }
-                        progress.report({message: 'Complete', work: {done: 100, total: 100}});
+                        progress.report({ message: 'Complete', work: { done: 100, total: 100 } });
                         this.dispose();
                         return;
                     }
-                    progress.report({message: 'Merging traces', work: {done: 70, total: 100}});
-                    const experiment = await this.experimentManager.openExperiment(this.uri.name, traces);
-                    if (experiment) {
-                        this.openedExperiment = experiment;
-                        this.title.label = 'Trace: ' + experiment.name;
-                        this.id = experiment.UUID;
+                    progress.report({ message: 'Merging traces', work: { done: 70, total: 100 } });
+
+                    if (traces === undefined || traces.length === 0) {
+                        this.messageService.error('Invalid trace(s): ' + invalidTraces.toString());
+                        this.dispose();
+                    } else {
+                        const experiment = await this.experimentManager.openExperiment(this.uri.name, traces);
+                        if (experiment) {
+                            this.openedExperiment = experiment;
+                            this.title.label = 'Trace: ' + experiment.name;
+                            this.id = experiment.UUID;
 
-                        if (this.isVisible) {
-                            TraceViewerWidget.widgetActivatedEmitter.fire(experiment);
+                            if (this.isVisible) {
+                                TraceViewerWidget.widgetActivatedEmitter.fire(experiment);
+                            }
                         }
+                        // Display the names of all the incorrect traces.
+                        if (Array.isArray(invalidTraces) && invalidTraces.length) {
+                            this.messageService.error('Invalid trace(s): ' + invalidTraces.toString());
```

Comment:
Since this line appears both in the `if` and `else` clauses, can it be extracted outside of the condition block? Then perhaps the whole `if` block can be slightly simplified.

Reply:
Agree. However, with the new proposed changes, this can't be extracted out :(.